packages/cactus-plugin-satp-hermes/src/test/typescript/integration/recovery.test.ts

```diff
@@ -0,0 +1,279 @@
+import "jest-extended";
+import { Secp256k1Keys } from "@hyperledger/cactus-common";
+import { CrashManager } from "../../../main/typescript/gol/crash-manager";
+import {
+  LocalLog,
+  SupportedChain,
+  GatewayIdentity,
+  Address,
+} from "../../../main/typescript/core/types";
+import {
+  AssetSchema,
+  CredentialProfile,
+  LockType,
+  SignatureAlgorithm,
+} from "../../../main/typescript/generated/proto/cacti/satp/v02/common/message_pb";
+import { v4 as uuidv4 } from "uuid";
+import { SATP_VERSION } from "../../../main/typescript/core/constants";
+import { SATPSession } from "../../../main/typescript/core/satp-session";
+import { getSatpLogKey } from "../../../main/typescript/gateway-utils";
+import { TokenType } from "../../../main/typescript/core/stage-services/satp-bridge/types/asset";
```

Comment:
Please add some tests:
crash recovery for stage 1 (feel free to provoke a crash at stage 1 ) (phase 1.1)
crash recovery for stage 2.1, 2.3A, 2.5 and 2.6
crash recovery for stage 3.2A, 3.3, 3.6A, 3.7
rollback for the stages and steps above
Please create test utility functions to help reduce code duplication. You have a good example here: packages/cactus-plugin-satp-hermes/src/test/typescript/test-utils.ts. Please add to this file. 
reference https://github.com/ietf-satp/draft-ietf-satp-core/blob/main/figures/message%20flow%20diagram/gateway-message-flow-asset-transfer-v20PNG.png